pkg/verifier/kube/kube_verifier.go

```diff
@@ -30,8 +68,278 @@ func NewKubeVerifier(clientset *kubernetes.Clientset, debug bool) (*KubeVerifier
 	}
 
 	return &KubeVerifier{
-		KubeClient: *kubeClient,
+		KubeClient: kubeClient,
 		Logger:     logger,
 		Output:     output.Output{},
 	}, nil
 }
+
+func (k *KubeVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.Output {
+	// Validate cloud platform type
+	if !vei.PlatformType.IsValid() {
+		vei.PlatformType = cloud.AWSClassic
+	}
+
+	if _, ok := vei.Probe.(curl.Probe); !ok {
+		return k.Output.AddError(errors.New("verification via pod mode only supports curl probe"))
+	}
+
+	if vei.Timeout <= 0 {
+		vei.Timeout = defaultCurlTimeout
+	}
+	k.writeDebugLogs(vei.Ctx, fmt.Sprintf("configured a %s timeout for each egress request", vei.Timeout))
+
+	// Generate egress lists for the given PlatformType
+	generatorVariables := map[string]string{"AWS_REGION": vei.AWS.Region}
+	generator := egress_lists.NewGenerator(vei.PlatformType, generatorVariables, k.Logger)
+	egressListStr, tlsDisabledEgressListStr, err := generator.GenerateEgressLists(vei.Ctx, vei.EgressListYaml)
+	if err != nil {
+		return k.Output.AddError(err)
+	}
+
+	// Generate curl commands
+	curlCommand, err := k.generateCurlCommands(egressListStr, tlsDisabledEgressListStr, vei.Timeout, vei.Proxy)
+	if err != nil {
+		return k.Output.AddError(err)
+	}
+
+	// We always want our Curl process to exit successfully, even if egress fails. This way, the pod succeeds, and then
+	// we parse the output to identify our failures like we would in EC2.
+	curlCommand = fmt.Sprintf("%s || true", curlCommand)
+
+	// Create and execute Job
+	jobName := fmt.Sprintf("osd-network-verifier-job-%d", time.Now().Unix())
+	proxySettings := k.buildProxyEnvironment(vei.Proxy)
+	resourceLimits := k.buildResourceRequirements()
+
+	jobInput := createJobInput{
+		JobName:                 jobName,
+		Namespace:               k.KubeClient.GetNamespace(),
+		ContainerImage:          defaultContainerImage,
+		CurlCommand:             curlCommand,
+		ProxySettings:           proxySettings,
+		TTLSecondsAfterFinished: defaultTTLSecondsAfterFinished,
+		ActiveDeadlineSeconds:   defaultActiveDeadlineSeconds,
+		BackoffLimit:            defaultBackoffLimit,
+		ResourceLimits:          resourceLimits,
+		Ctx:                     vei.Ctx,
+	}
+
+	err = k.createAndExecuteJob(jobInput)
+	if err != nil {
+		// Best effort delete the job
+		_ = k.KubeClient.DeleteJob(jobInput.Ctx, jobInput.JobName)
+		return k.Output.AddError(err)
+	}
+
+	// Collect and parse output
+	err = k.collectAndParseJobOutput(vei.Ctx, jobName, vei.Probe)
+	if err != nil {
+		k.Output.AddError(err)
+	}
+
+	// Best effort delete the job
+	_ = k.KubeClient.DeleteJob(jobInput.Ctx, jobInput.JobName)
```

Comment:
Any reason not to use `CleanupJob()` instead of `DeleteJob()`?

Reply:
I don't really think we need to block on deletion of the Job, however updated to use `CleanupJob`. I think I was mostly just looking for what was implemented and found `DeleteJob` did what I wanted. If the intention is to only expose `CleanupJob` then we could make `DeleteJob` a private function, and leave the caller only the one option.